prboom2/src/hexen/sv_save.c

```diff
@@ -217,7 +217,7 @@ static uint64_t SV_ReadFlags(void)
 
 static void *SV_ReadPtr(void)
 {
-    return (void *) (intptr_t) SV_ReadLong();
```

Comment:
Why this?

Reply:
this triggers the `-Wbad-function-cast` warning, because we are casting rom a signed value to a void*.
But now that you say it maybe the value needs to be sign-extended when going from a 32bit int to a 64bit void* ?

I should investigate first why there is this crazy int->void* cast in the first place.

Reply:
This code is actually "dead" and has no practical impact on execution I think. If you check the associated values they are read and then set to NULL anyway. I think all this code was autogenerated for chocolate hexen and then specific fields were patched to NULL. The read / write of pointers in sv_save can just be purged from the code. I guess chocolate hexen left it all in for save file compatibility.

Reply:
> I should investigate first why there is this crazy int->void* cast in the first place.

That has some historic background. Back in the days, HEXEN.EXE used to save entire `struct`s in its savegames. These structs contained pointers, which were of course invalid once the savegame was restored, thus they were read back into the struct but then invalidated. To improve portability to systems where `sizeof void* != sizeof int`, the writing and reading of the struct content was split into several functions, one for each struct member. So, in order to restore a pointer element of a struct from a savegame, you'd actually have to read an `int` and convert this into a pointer. However, some compiled complain if you cast an `int` into a pointer of different size and thus the intermediate cast had to be added. The return value is still invalidated, after all.

Reply:
Thanks for this info, I love this kind of historic context.